### ISSUE TITLE
Feature: Add error codes to concrete API 400_BAD_REQUEST responses

### ISSUE THREAD

**Issue by KhaledBousrih** [Netsach/concrete-datastore#23]
When Concrete API responds with a 400_BAD_REQUEST, the response should be:
```python
return Response(
    data={
        "message": error_data,
        "_errors": [error_code],
    },
    status=HTTP_400_BAD_REQUEST,
```

Here is a list of the missing error codes with the corresponding cases:

-  `"_errors": ["INVALID_DATA"]`
    +  serializer errors
    +  change password: user not found

-  `"_errors": ["WRONG_EMAIL_ADDRESS"]`
    +  retrieve secure token: user not found
    +  generate secure token: user not found

-  `"_errors": ["INVALID_TOKEN"]`
    +  secure login: token does not exist
    +  change password: token does not exist
    +  change password: token is not the token's user
    +  change password: token expired

-  `"_errors": ["TOKEN_HAS_EXPIRED"]`
    +  secure login: token has expired

-  `"_errors": ["CANNOT_USE_SAME_PASSWORD"]`
    +  change password: same password as the old one

-  `'_errors': ['INVALID_PARAMETER']`
    +  register: anonymous user passes an `email_format`
    +  register: email_format has onvalid format

-  `'_errors': ['MISMATCH_PASSWORDS']`
    +  register: password1 != password2
    +  change password: password1 != password2

-  `"_errors": ["EMAIL_NOT_AUTHORIZED_TO_REGISTER"]`
    +  register: email not allowed to register

-  `"_errors": [e.code]`  : this contains the PasswordSecurityException message
    +  password security

-  `'_errors': ['INVALID_QUERY']`
    +  CRUD queries: if timestamp_start < 0
    +  CRUD queries: if c_resp_page_size is not or is < 1
    +  CRUD queries: if c_resp_nested is not in [true`, `false`]
    +  CRUD queries: if wrong date format
    +  CRUD queries: if filter against not allowed field

-  `'_errors': ['INVALID_SCOPES_HEADERS']`
    +  CRUD scope: invalid entity UID

**Comment by paschembri** [comment]
For the invalid scope, it would be better (more generic) to name this error code `INVALID_SCOPE_HEADERS`

**Comment by lcognat** [comment]
> For the invalid scope, it would be better (more generic) to name this error code `INVALID_SCOPE_HEADERS`

I would prefer the use of the plural : `INVALID_SCOPES_HEADERS`

**Comment by KhaledBousrih** [comment]
Fixed in `v1.11.0`
